src/Pos/Communication/Limits.hs

```diff
@@ -209,9 +228,9 @@ instance MessageLimited (DataMsg GtMsgContents) where
     getMsgLenLimit _ =
         return $ each %~ fmap DataMsg $
             ( mcCommitmentMsgLenLimit
-            , noLimit
+            , NoLimit
```

Comment:
1. I am afraid it's bad. `Opening` wraps `ByteString` which can potentially be infinite. So AFAIU we may parse 100500 bytes when we receive an `Opening`.
2. In general I don't fully understand why we need this type.
2.1. If we are dealing with type which has a limited size by structure (e. g. `Word8`), why can't we just use this limit?
2.2. `NoLimit` is dangerous, because even if we think that type has a limited size by structure, it can happen that we missed something (like you apparently missed it for `Opening`).